beacon-chain/rpc/eth/beacon/handlers.go

```diff
@@ -640,6 +641,132 @@ func (bs *Server) validateEquivocation(blk interfaces.ReadOnlyBeaconBlock) error
 	return nil
 }
 
+// ListBlockHeaders retrieves block headers matching given query. By default it will fetch current head slot blocks.
+func (bs *Server) ListBlockHeaders(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "beacon.ListBlockHeaders")
+	defer span.End()
+
+	rawSlot := r.URL.Query().Get("slot")
+	rawParentRoot := r.URL.Query().Get("parent_root")
+
+	var err error
+	var blks []interfaces.ReadOnlySignedBeaconBlock
+	var blkRoots [][32]byte
+
+	if rawParentRoot != "" {
+		if valid := shared.ValidateHex(w, "Parent Root", rawParentRoot); !valid {
+			return
+		}
+		parentRoot, err := hexutil.Decode(rawParentRoot)
+		if err != nil {
+			errJson := &http2.DefaultErrorJson{
```

Comment:
We have a `HandleError` method that will make the code more concise. Same comment for other usages of the error struct.